pkg/util/glog/glog.go

```diff
@@ -1,101 +1,234 @@
 }
 
-// gverbose handles glog.V(x) calls
-type gverbose struct {
-	glog.Verbose
+func (glogger) Info(args ...interface{}) {
+	log.InfoDepth(2, args...)
 }
 
-func (gverbose) Is(level int) bool {
-	return bool(glog.V(glog.Level(level)))
+func (glogger) Warningf(format string, args ...interface{}) {
+	log.WarningDepth(2, fmt.Sprintf(format, args...))
 }
 
-func (gverbose) V(level int) Logger {
-	if glog.V(glog.Level(level)) {
-		return Log
-	}
-	return None
+func (glogger) Warning(args ...interface{}) {
+	log.WarningDepth(2, args...)
+}
+
+func (glogger) Errorf(format string, args ...interface{}) {
+	log.ErrorDepth(2, fmt.Sprintf(format, args...))
+}
+
+func (glogger) Error(args ...interface{}) {
+	log.ErrorDepth(2, args...)
+}
+
+func (glogger) Fatalf(format string, args ...interface{}) {
+	log.FatalDepth(2, fmt.Sprintf(format, args...))
+}
+
+func (glogger) Fatal(args ...interface{}) {
+	log.FatalDepth(2, args...)
+}
+
+// gverbose handles glog.V(x) calls.
+type gverbose struct {
+	log.Verbose
 }
 
 func (g gverbose) Infof(format string, args ...interface{}) {
 	g.Verbose.Infof(format, args...)
 }
 
-// file logs the provided messages at level or below to the writer, or delegates
+func (g gverbose) Info(args ...interface{}) {
+	g.Verbose.Info(args...)
+}
+
+// FileLogger logs the provided messages at level or below to the writer, or delegates
 // to glog.
-type file struct {
-	w     io.Writer
-	level int
+type FileLogger struct {
+	w     *bufio.Writer
+	level int32
 }
 
-func (f file) Is(level int) bool {
+// Is returns whether the current logging level is greater than or equal to the parameter.
+func (f *FileLogger) Is(level int32) bool {
 	return level <= f.level
 }
 
-func (f file) V(level int) Logger {
-	// only log things that glog allows
-	if !glog.V(glog.Level(level)) {
-		return None
-	}
-	// send anything above our level to glog
-	if level > f.level {
-		return Log
+// V will returns a logger which will discard output if the specified level is greater than the current logging level.
+func (f *FileLogger) V(level int32) VerboseLogger {
+	// Is the loglevel set verbose enough to accept the forthcoming log statement
+	if log.V(log.Level(level)) {
+		return f
 	}
-	return f
+	// Otherwise discard
+	return None
 }
 
-func (f file) Infof(format string, args ...interface{}) {
-	fmt.Fprintf(f.w, format, args...)
-	if !strings.HasSuffix(format, "\n") {
-		fmt.Fprintln(f.w)
+type severity int32
+
+const (
+	infoLog severity = iota
+	warningLog
+	errorLog
+	fatalLog
+)
+
+// Elevated logger methods output a detailed prefix for each logging statement.
+// At present, we delegate to glog to accomplish this.
+type elevated func(int, ...interface{})
+
+type severityDetail struct {
+	prefix     string
+	delegateFn elevated
+}
+
+var severities = []severityDetail{
+	infoLog:    {"", log.InfoDepth},
+	warningLog: {"WARNING: ", log.WarningDepth},
+	errorLog:   {"ERROR: ", log.ErrorDepth},
+	fatalLog:   {"FATAL: ", log.FatalDepth},
+}
```

Comment:
reorder this block so the entire FileLogger struct + method implementations are grouped together.